server/controllers/userController.js

```diff
@@ -15,19 +15,52 @@ async function addUser(req, res) {
       email,
       password,
     } = req.body;
+
     try {
-      const newUser = await user.create({
-        firstName,
-        lastName,
-        email,
-        password,
-      });
-      res.status(201);
-      res.send({ message: 'Successfully created new user', _id: newUser._id }); // eslint-disable-line
+      const preExistingUser = await user.findOne({email, password})
```

Comment:
You probably don't need to look for the password, just the email should be fine. Otherwise a user could sign up with the same email using different passwords